neural_network/convolution_neural_network.py

```diff
@@ -135,7 +135,7 @@ def convolute(self, data, convs, w_convs, thre_convs, conv_step):
             )
             data_featuremap.append(featuremap)
 
-        # expanding the data slice to One dimenssion
+        # expanding the data slice to One dimension
```

Comment:
```suggestion
        # expanding the data slice to one dimension
```
Not strictly speaking a typo, but it's weird to have a single word capitalized for no good reason